File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/Proiect_MPP/service/main/IMainService.cs
using Proiect_MPP.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proiect_MPP.service.main
{
    public interface IMainService
    {
        void updateFlight(Flight flight, int id);

        List<Flight> findByDestinationDate(string destination, DateTime date);

        Airport? findAirportById(int id);

        Flight? findFlightById(int id);

        List<Flight> getAllAvailableFlights();
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/Thrift/src/main/java/root/proiect_mpp/service/main/IMainService.java
package root.proiect_mpp.service.main;

import root.proiect_mpp.domain.Airport;
import root.proiect_mpp.domain.Flight;

import java.time.LocalDate;
import java.util.Collection;

public interface IMainService {
    void updateFlight(Flight elem, int id);

    Collection<Flight> findByDestinationDate(String destination, LocalDate date);

    Airport findAirportById(int id);

    Flight findFlightById(int id);

    Collection<Flight> getAllAvailableFlights();
}
